pontoon/base/vcs_models.py

```diff
@@ -0,0 +1,107 @@
+import os.path
+from itertools import chain
+
+
+class VCSProject(object):
+    def __init__(self, db_project):
+        self.db_project = db_project
+
+        self.resources = {}
+        for path in db_project.relative_source_paths():
+            self.resources[path] = VCSResource(self, path)
+
+    @property
+    def entities(self):
+        return chain.from_iterable(
+            resource.entities.values() for resource in self.resources.values()
+        )
+
+
+class VCSResource(object):
+    def __init__(self, vcs_project, relative_path):
+        from pontoon.base import formats
+
+        self.vcs_project = vcs_project
+        self.relative_path = relative_path
+        self.files = {}
+        self.entities = {}
+
+        db_project = vcs_project.db_project
+        for locale in db_project.locales.all():
+            resource_path = os.path.join(
+                db_project.checkout_path,
+                db_project.locale_directory_name(locale.code),
+                self.relative_path
+            )
+            try:
+                resource_file = formats.parse(resource_path)
+            except IOError:
+                continue  # File doesn't exist, let's move on
+
+            self.files[locale.code] = resource_file
+            for index, format_entity in enumerate(resource_file.entities):
+                if format_entity.key in self.entities:
+                    vcs_entity = self.entities[format_entity.key]
+                else:
+                    vcs_entity = VCSEntity(
+                        resource=self,
+                        string=format_entity.string,
+                        comments=format_entity.comments,
+                        order=index
+                    )
+                    self.entities[format_entity.key] = vcs_entity
+
+                vcs_entity.translations[locale.code] = VCSTranslation(format_entity)
+
+    def save(self):
+        for locale_code, resource_file in self.files.items():
+            resource_file.save()
+
+
+class VCSEntity(object):
+    def __init__(self, resource, string, comments, order):
+        self.resource = resource
+        self.string = string
```

Comment:
Yeah, the plural support here is half-assed for now and will be fleshed out when we add a second format that supports proper plurals.